app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,59 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all
+  end
+
+  def upvote
+    @album = Album.find(params[:id])
+    @album.update_attribute(:ranking, @album.ranking + 1)
+    redirect_to action: :show
+  end
+
+  def show
+    @album = Album.find(params[:id])
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(title: album_params[:album][:title],
```

Comment:
Because we're using the special "strong params" syntax to implement `album_params` we can actually pass its return value directly to `create`:
```ruby
@album = Album.create(album_params[:album]) do |album|
  album.ranking = 0
end
```
That way, if we want to add more parameters to Album (and Movie or Book) we just need to add them into `album_params` and everything else will work itself out.